lib/monad.js

```diff
@@ -0,0 +1,72 @@
+'use strict';
+
+// id :: a -> a
+function id(x) {
+  return x;
+}
+
+// curry :: ((a, b) -> c) -> a -> b -> c
+function curry(f, ...args1) {
+  return (...args2) => f(...args1, ...args2);
+}
+
+function withCb(func) {
+  return (...argsCb) => {
+    const callback = argsCb[argsCb.length - 1];
+    const args = argsCb.slice(0, argsCb.length - 1);
+    return func(args, callback);
+  };
+}
+
+// fmap :: Functor f => (a -> b) -> f a -> f b
+function fmap(f, asyncF) {
+  const asyncF2 = withCb((args, callback) => of(f(...args))(callback));
+  return chain(asyncF, asyncF2);
+}
+
+// pure :: Applicative f => f (a -> b) -> f a -> f b
+function of(...args) {
```

Comment:
Because `of` creates asynchronous function but doesn't call it. The main purpose is to use it in chaining.
```javascript
const fakeReading = of('My text which is not downloaded from file');
const realReading = curry(fs.readFile, 'myFile.txt');

chain(fakeReading, curry(fs.writeFile, 'myFile.txt.cpy'))(api.common.emptiness);
```
Here we can change only variable name and way of reading will change. Unlike there:
```javascript
const fakeText = 'My text which is not downloaded from file';
fs.writeFile('myFile.txt.cpy', fakeText, api.common.emptiness);

fs.readFile('myFile.txt', (err, realText) =>
  fs.writeFile('myFile.txt.cpy', realText, api.common.emptiness)
);
```